src/logic/addons.h

```diff
@@ -91,16 +93,20 @@ struct AddOnInfo {
 	bool verified;  // Only valid for Remote add-ons
 
 	AddOnFileList file_list;  // Get rid of this ASAP
+	std::map<std::string /* name */, std::string /* description */> screenshots;
 
 	uint32_t total_file_size;     // total size of all files, in bytes
 	std::string upload_username;  // who uploaded (may be different from author)
 
 	// TODO(Nordfriese): These are not yet implemented on the server-side
 	std::time_t upload_timestamp;  // date and time when this version was uploaded
 	uint32_t download_count;       // total times downloaded
-	uint32_t votes;                // total number of votes
-	float average_rating;          // average rating between 1.0 and 10.0 (0 if no votes)
+	std::map<uint8_t /* rating 1..10 */, uint32_t /* number of votes */>
+	   votes;  // total number of votes for each of the ratings 1-10
```

Comment:
Can't this just be a simple array (votes[0]...votes[9]) ? A map seems a bit overkill to me here.